api/server/server.go

```diff
@@ -1137,8 +1137,17 @@ func (s *Server) getContainersByName(version version.Version, w http.ResponseWri
 		return fmt.Errorf("Missing parameter")
 	}
 
+	if err := parseForm(r); err != nil {
+		return err
+	}
+
+	no_err := r.Form.Get("no_err")
 	containerJSON, err := s.daemon.ContainerInspect(vars["name"])
 	if err != nil {
+		if strings.Contains(strings.ToLower(err.Error()), "no such id") && no_err == "1" {
+			w.WriteHeader(http.StatusNoContent)
```

Comment:
I don't like it either, but didn't find a better way, hack 404 seems even more hacky. I'm open to change it if someone has better idea :)